doc/jenkins.rst

```diff
@@ -0,0 +1,70 @@
+Integration with Jenkins
+========================
+
+`Universum` is usually launched by `Jenkins` as one long step in a single build stage. Because of that,
+the whole `Universum` log is printed as a plain text without navigation anchors.
+
+.. admonition:: Compare with these:
+
+    * When ran locally, all Universum build step logs are stored to separate files instead of printing their
+      output to console
+    * When launched by TeamCity, `Unviersum` uses `service messages
+      <https://www.jetbrains.com/help/teamcity/service-messages.html>`__ to increase log readability
+
+To simplify integration and debug, we provide a user-friendly interactive log with collapsible blocks and
+other features. This log is generated when `--html-log <args.html#Output>`__ option is passed to command line.
+It can be accessed from project `artifacts <args.html#Artifact\ collection>`__.
+
+.. note::
+
+    By default, Jenkins `does not render interactive content <https://www.jenkins.io/doc/book/security/user-content/>`__.
+    This means that without changing server settings, interactive features of the generated log will be
+    inaccessible when opened directly from Jenkins artifacts.
+
+
+Jenkins Content-Security-Policy
+-------------------------------
+
+A recommended way to allow Jenkins server to render interactive user content is to `configure Jenkins to publish
+artifacts on a different domain <https://www.jenkins.io/doc/book/security/user-content/#resource-root-url>`__
+by changing ``Resource Root URL`` in `Manage Jenkins » Configure System » Serve resource files from another domain`
+from something like ``my.jenkins.io`` to ``res.my.jenkins.io``.
+
+Note that Jenkins interaction with resource domain, located on the same host is not done via ``localhost``
+network interface. Jenkins treats ``another domain`` as an external URL when redirecting. This means that:
+
+1. Both ``my.jenkins.io`` and ``res.my.jenkins.io`` domain names must be resolved correctly
+2. All headers must be passed correctly
+
+.. note::
+
+    If main server domain name is not resolved using DNS, ``/etc/hosts`` or any other means, the web-interface
+    will only be accessible via IP, and not the name. As resource domain might be located at the same IP,
+    the resolving of both names is crucial
+
+Here are the symptoms of domain names not resolving correctly:
+
+1. Jenkins warnings when trying to save the updated settings
+2. Client inability to access said pages (timeout error)
+
+A reason for headers not passed correctly can be Nginx configuration. Without specification Nginx replaces
```

Comment:
This is the first time the doc mentions nginx.
However, from the paragraph it looks like it assumes the reader already knows why we use nginx.
I suggest to mention nginx in the intro section on this page.
We can assume that it is present and setup as a reverse proxy, but have to declare it explicitly.